pkg/test/fakes/pb/controller.go

```diff
@@ -0,0 +1,106 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package pb
+
+import (
+	"context"
+
+	"github.com/gogo/protobuf/proto"
+	"google.golang.org/grpc"
+)
+
+type settings map[string]string
+
+func newSettings() settings {
+	return settings(make(map[string]string))
+}
+
+func (s settings) setBool(name string, value bool) settings {
+	return s.setString(name, "true")
+}
+
+func (s settings) setString(name string, value string) settings {
+	if s == nil {
```

Comment:
Can `s` ever be `nil`? We're calling a method on `s` ... seems strange, but it may be a go-ism that I'm not familiar with.

Reply:
I had a slightly different model in mind when I started. Removed.